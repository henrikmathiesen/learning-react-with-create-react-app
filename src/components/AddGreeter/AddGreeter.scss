@import '../../_settings';

.AddGreeter {
    border: $component-border;
    text-align: center;
    background: $belize-blue;
    color: $almost-black;
    margin: $gutter-md;
    padding: $gutter-md;
}
