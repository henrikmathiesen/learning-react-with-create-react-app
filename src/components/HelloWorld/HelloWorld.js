import React from 'react';
import './HelloWorld.css';

const HelloWorld = (props) => {
    return(
        <div className="HelloWorld" onClick={props.test}>Hello {props.name} World!</div>
    );
}

export default HelloWorld;



// HelloWorld component, can recieve a prop that is passed in from parent.
// Using the function style since this component focuses on rendering markup and props.
// A function style component does not need a render function it just returns JSX.
