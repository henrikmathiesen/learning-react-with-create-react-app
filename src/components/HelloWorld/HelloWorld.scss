@import '../../_settings';

.HelloWorld {
    border: $component-border;
    text-align: center;
    background: $emerald;
    color: $almost-black;
    margin: $gutter-md;
    padding: $gutter-md;
}
