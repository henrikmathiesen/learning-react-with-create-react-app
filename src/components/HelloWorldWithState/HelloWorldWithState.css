.HelloWorldWithState {
    border: 2px solid black;
    text-align: center;
    background: #3498db;
    color: black;
    color: #333;
    margin: 20px;
    padding: 20px;
}
