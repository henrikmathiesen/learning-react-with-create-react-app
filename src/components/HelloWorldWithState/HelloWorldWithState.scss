@import '../../_settings';

.HelloWorldWithState {
    border: 2px solid $black;
    text-align: center;
    background: $river;
    color: $almost-black;
    margin: 20px;
    padding: 20px;
}

@import 'foo';
