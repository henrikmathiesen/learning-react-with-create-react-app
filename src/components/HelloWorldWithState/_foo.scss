// dont need to import _settings since parent sass file has it imported

.HelloWorldWithState {
    color:$sun-flower;
}

.testing {
    columns: 100px 4;
}

.testing-too {
    color:pink;
}
