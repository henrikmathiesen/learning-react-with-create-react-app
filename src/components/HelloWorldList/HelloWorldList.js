import React, { Component } from 'react';
import HelloWorldWithState from '../HelloWorldWithState/HelloWorldWithState';

class HelloWorldList extends Comment {
    render() {
        return (
            <div className="HelloWorldList">
                <HelloWorldWithState name="Adam" />
                <HelloWorldWithState name="Bertil" />
            </div>
        );
    }
}

export default HelloWorldList;
