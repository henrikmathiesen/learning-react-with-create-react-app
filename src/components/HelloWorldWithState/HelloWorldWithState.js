import React, { Component } from 'react';
import './HelloWorldWithState.css';

class HelloWorldWithState extends Component {

    constructor(props) {
        super(props);
        this.state = { greeting: 'Hello there' };

        this.french = this.french.bind(this); // *
    }

    french() {
        this.setState({ greeting: 'Bonjour' });
    }

    render() {
        return (
            <div className="HelloWorldWithState">
                <div>
                    {this.state.greeting} {this.props.name} World!
                </div>
                <div>
                    <button onClick={this.french}>Frenchify!</button>
                </div>
            </div>
        );
    }
}

export default HelloWorldWithState;


// Import the default export from 'react' in node_modules as React.
// Also import named export Component from 'react'.
// We use a class here, because this component focuses on more than just rendering content.
// We need to define a render function in our class for rendering JSX.
// This component can also get props passed to it, but since this is a class we get those in the constructor,
// that means that we have to use this.props.name instead of props.name

// OBS
// A simple constructor like this is not needed, it seems that it is done automaticly, and in fact the linter complains about this as Useless constructor
// constructor(props) {
//  super(props);
// }

// HOWEVER
// If we use state, we set that up in the constructor, lite above

// Changing state
// Any state changes NEED to happen via the this.setState function

// * This tells Javascript "Hey, any time you see a reference to this inside of the frenchify function, I want you specifically to refer to ME."
