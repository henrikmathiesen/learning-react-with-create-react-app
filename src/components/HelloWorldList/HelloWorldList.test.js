import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorldList from './HelloWorldList';
import AddGreeter from '../AddGreeter/AddGreeter';
import HelloWorldWithState from '../HelloWorldWithState/HelloWorldWithState';
import HelloWorld from '../HelloWorld/HelloWorld';

describe('HelloWorldList', () => {

    const component = shallow(
        <HelloWorldList />
    );

    // 1) Write a generic snapshot test

    it('Renders and matches our snapshot', () => {
        const component = renderer.create(
            <HelloWorldList />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    // 2) Write some sub-component-specific tests

    it('contains an AddGreeter subcomponent', () => {
        expect(component.find(AddGreeter)).toHaveLength(1);
        expect(component.find(HelloWorld).length).toBe(1);  // jasmine syntax works here also (toEqual also)
    });

});
