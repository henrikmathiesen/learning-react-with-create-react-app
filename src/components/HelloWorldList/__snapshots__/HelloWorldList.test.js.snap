exports[`HelloWorldList Renders and matches our snapshot 1`] = `
<div>
  <div
    className="AddGreeter">
    <input
      onChange={[Function]}
      type="text"
      value="" />
      
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
  <div
    className="HelloWorldWithState">
    <div>
      Hello there
       
      Adam
       World!
    </div>
    <div>
      <button
        className="ts-french"
        onClick={[Function]}>
        Frenchify!
      </button>
      <button
        className="ts-remove"
        onClick={[Function]}>
        Remove
      </button>
    </div>
  </div>
  <div
    className="HelloWorldWithState">
    <div>
      Hello there
       
      Bertil
       World!
    </div>
    <div>
      <button
        className="ts-french"
        onClick={[Function]}>
        Frenchify!
      </button>
      <button
        className="ts-remove"
        onClick={[Function]}>
        Remove
      </button>
    </div>
  </div>
  <div
    className="HelloWorldWithState">
    <div>
      Hello there
       
      Ceasar
       World!
    </div>
    <div>
      <button
        className="ts-french"
        onClick={[Function]}>
        Frenchify!
      </button>
      <button
        className="ts-remove"
        onClick={[Function]}>
        Remove
      </button>
    </div>
  </div>
  <div
    className="HelloWorld"
    onClick={[Function]}>
    Hello 
     World!
  </div>
</div>
`;
